From dd418384171473c3e31386a1b4792f8c555dc744 Mon Sep 17 00:00:00 2001
From: Jiri Dostal <jdostal@redhat.com>
Date: Wed, 22 Jul 2015 13:19:34 +0200
Subject: [PATCH] 1181152 - XSS when altering user details and going somewhere
 where you are choosing user         - Escaped tags in real names

---
 backend/server/rhnPackage.py                                    | 2 +-
 java/code/webapp/WEB-INF/pages/admin/multiorg/org_users.jsp     | 2 +-
 java/code/webapp/WEB-INF/pages/admin/multiorg/sat_org_users.jsp | 2 +-
 java/code/webapp/WEB-INF/pages/admin/users/activelist.jsp       | 2 +-
 .../webapp/WEB-INF/pages/common/fragments/manage/managers.jspf  | 2 +-
 .../WEB-INF/pages/common/fragments/manage/subscribers.jspf      | 2 +-
 .../WEB-INF/pages/common/fragments/user/userlist_columns.jspf   | 2 +-
 java/code/webapp/WEB-INF/pages/groups/adminlist.jsp             | 2 +-
 8 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/backend/server/rhnPackage.py b/backend/server/rhnPackage.py
index fd9cd0fa8ee..b8ee26daa6f 100644
--- a/backend/server/rhnPackage.py
+++ b/backend/server/rhnPackage.py
@@ -203,7 +203,7 @@ def get_info_for_package(pkg, channel_id, org_id):
               'channel_id': channel_id,
               'org_id': org_id}
     # yum repo has epoch="0" not only when epoch is "0" but also if it's NULL
-    if pkg[3] == '0' or pkg[3] == '':
+    if pkg[3] == '0' or pkg[3] == '' or pkg[3]==None:
         epochStatement = "(epoch is null or epoch = :epoch)"
     else:
         epochStatement = "epoch = :epoch"
diff --git a/java/code/webapp/WEB-INF/pages/admin/multiorg/org_users.jsp b/java/code/webapp/WEB-INF/pages/admin/multiorg/org_users.jsp
index 2de867c8426..d76fcb087c3 100644
--- a/java/code/webapp/WEB-INF/pages/admin/multiorg/org_users.jsp
+++ b/java/code/webapp/WEB-INF/pages/admin/multiorg/org_users.jsp
@@ -78,7 +78,7 @@
                    sortable="false"
                    headerkey="realname.displayname"
                    attr="userLastName">
-                <c:out value="${current.userDisplayName}" escapeXml="false" />
+                <c:out value="${current.userDisplayName}" escapeXml="true" />
         </rl:column>
         <rl:column bound="false"
                    sortable="false"
diff --git a/java/code/webapp/WEB-INF/pages/admin/multiorg/sat_org_users.jsp b/java/code/webapp/WEB-INF/pages/admin/multiorg/sat_org_users.jsp
index e640491a7eb..5880d441153 100644
--- a/java/code/webapp/WEB-INF/pages/admin/multiorg/sat_org_users.jsp
+++ b/java/code/webapp/WEB-INF/pages/admin/multiorg/sat_org_users.jsp
@@ -55,7 +55,7 @@
                    sortable="false"
                    headerkey="realname.displayname"
                    attr="userLastName">
-                <c:out value="<a href=\"mailto:${current.address}\">${current.userLastName}, ${current.userFirstName}</a>" escapeXml="false"/>
+                <c:out value="<a href=\"mailto:${current.address}\">${current.userLastName}, ${current.userFirstName}</a>" escapeXml="true"/>
         </rl:column>
         <rl:column bound="false"
                    sortable="false"
diff --git a/java/code/webapp/WEB-INF/pages/admin/users/activelist.jsp b/java/code/webapp/WEB-INF/pages/admin/users/activelist.jsp
index dd91fa5844c..8fade0f9349 100644
--- a/java/code/webapp/WEB-INF/pages/admin/users/activelist.jsp
+++ b/java/code/webapp/WEB-INF/pages/admin/users/activelist.jsp
@@ -44,7 +44,7 @@
                    sortable="true"
                    headerkey="realname.displayname"
                    sortattr="userLastName">
-                <c:out value="${current.userLastName}, ${current.userFirstName}" />
+                <c:out escapeXml="true" value="${current.userLastName}, ${current.userFirstName}" />
         </rl:column>
 
         <!--  Roles column -->
diff --git a/java/code/webapp/WEB-INF/pages/common/fragments/manage/managers.jspf b/java/code/webapp/WEB-INF/pages/common/fragments/manage/managers.jspf
index 619471bf54b..1bec08c6bfb 100644
--- a/java/code/webapp/WEB-INF/pages/common/fragments/manage/managers.jspf
+++ b/java/code/webapp/WEB-INF/pages/common/fragments/manage/managers.jspf
@@ -39,7 +39,7 @@
          <rl:column sortable="false"
             bound="false"
             headerkey="realname.displayname" >
-            ${current.userLastName}, ${current.userFirstName}
+            <c:out escapeXml="true" value="${current.userLastName},${current.userFirstName}" />
          </rl:column>
 
          <rl:column sortable="false"
diff --git a/java/code/webapp/WEB-INF/pages/common/fragments/manage/subscribers.jspf b/java/code/webapp/WEB-INF/pages/common/fragments/manage/subscribers.jspf
index 0ac6b3e9614..24ed0e1d111 100644
--- a/java/code/webapp/WEB-INF/pages/common/fragments/manage/subscribers.jspf
+++ b/java/code/webapp/WEB-INF/pages/common/fragments/manage/subscribers.jspf
@@ -39,7 +39,7 @@
          <rl:column sortable="false"
             bound="false"
             headerkey="realname.displayname" >
-            ${current.userLastName}, ${current.userFirstName}
+            <c:out escapeXml="true" value="${current.userLastName},${current.userFirstName}" />
          </rl:column>
 
          <rl:column sortable="false"
diff --git a/java/code/webapp/WEB-INF/pages/common/fragments/user/userlist_columns.jspf b/java/code/webapp/WEB-INF/pages/common/fragments/user/userlist_columns.jspf
index 27944d08cc1..f14bf856d5a 100644
--- a/java/code/webapp/WEB-INF/pages/common/fragments/user/userlist_columns.jspf
+++ b/java/code/webapp/WEB-INF/pages/common/fragments/user/userlist_columns.jspf
@@ -4,7 +4,7 @@
         sortable="true"
         headerkey="realname.displayname"
         sortattr="userLastName">
-        <c:out value="${current.userLastName}, ${current.userFirstName}" />
+        <c:out escapeXml="true" value="${current.userLastName}, ${current.userFirstName}" />
 </rl:column>
 
 <rl:column bound="true"
diff --git a/java/code/webapp/WEB-INF/pages/groups/adminlist.jsp b/java/code/webapp/WEB-INF/pages/groups/adminlist.jsp
index dc5a606c135..33333f55823 100644
--- a/java/code/webapp/WEB-INF/pages/groups/adminlist.jsp
+++ b/java/code/webapp/WEB-INF/pages/groups/adminlist.jsp
@@ -50,7 +50,7 @@
          <rl:column sortable="false"
             bound="false"
             headerkey="realname.displayname" >
-            ${current.userLastName}, ${current.userFirstName}
+            <c:out escapeXml="true" value="${current.userLastName},${current.userFirstName}" />
          </rl:column>
 
          <rl:column sortable="true"
